examples/java-matter-controller/java/src/com/matter/controller/commands/common/Command.kt

```diff
@@ -0,0 +1,256 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package com.matter.controller.commands.common
+
+import java.util.ArrayList
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicLong
+
+/**
+ * @brief Matter Controller command
+ * @details The base class of all the commands the Matter Controller supports, which are actions
+ * that may be performed. Commands are verb-like, such as pair a Matter device or discover
+ * Matter devices from the environment.
+ */
+abstract class Command(
+  private val name: String,
+  private val helpText: String? = null) {
+  private val arguments = ArrayList<Argument>()
+
+  fun getName(): String {
+    return name
+  }
+
+  fun getHelpText(): String? {
+    return helpText
+  }
+
+  fun getArgumentName(index: Int): String {
+    return arguments[index].name
+  }
+
+  fun getArgumentsCount(): Int {
+    return arguments.size
+  }
+
+  fun getArgumentIsOptional(index: Int): Boolean {
+    return arguments[index].isOptional
+  }
+
+  /**
+   * @brief Get attribute argument if it exists, there is at most one Attribute argument per command
+   * @return A pointer to an Optional where the Attribute argument will be stored
+   */
+  fun getAttribute(): String? {
+    return arguments.find { arg -> arg.type === ArgumentType.ATTRIBUTE }?.value as? String
+  }
+
+  /**
+   * @brief Get argument description if it exists
+   * @return A pointer to an Optional where the argument description will be stored
+   */
+  fun getArgumentDescription(index: Int): String? {
+    return arguments[index].desc
+  }
+
+  fun addArgumentToList(arg: Argument) {
```

Comment:
For the `addArgument` functions, it might be nice to return a reference that can be used for chaining. E.g. 

```
command.addArgumentToList("arg 1")
   .addArgumentToList("arg 2")
```

Reply:
Yes, but we don't have this use case, this function is only called once within each addArgument fun 

```
  fun addArgument(
    name: String?, out: AtomicBoolean?, desc: String?, optional: Boolean
  ) {
    val arg = Argument(name!!, out!!, desc, optional)
    addArgumentToList(arg)
  }
```